Mage/src/main/java/mage/abilities/effects/common/continuous/SetBasePowerToughnessSourceEffect.java

```diff
@@ -107,4 +106,96 @@ public boolean apply(Game game, Ability source) {
         }
         return true;
     }
+
+    void setStaticText(Duration duration) {
+
+        if (power == null){
+            staticText = getSingleStatText(toughness, "toughness", duration);
+            return;
+        }
+        if (toughness == null){
+            staticText = getSingleStatText(power, "power", duration);
+            return;
+        }
+
+        StringBuilder sb = new StringBuilder("{this}");
+
+        boolean indefiniteDuration = duration.toString().isEmpty() || duration == Duration.EndOfGame;
+
+        if (indefiniteDuration) {
+            sb.append("'s power ");
+        } else {
+            sb.append(" has base power ");
+        }
+
+        // Relies on StaticValue messages all being empty
+        String powerMessage = power.getMessage(ValuePhrasing.EQUAL_TO);
+        String toughnessMessage = toughness.getMessage(ValuePhrasing.EQUAL_TO);
+
+        if (toughnessMessage.equals(powerMessage)) {
+            sb.append("and toughness ");
+        }
+
+        if (power instanceof StaticValue) {
+            // Assume that if one is static, the other is static too
```

Comment:
is this enforced by constructor? if so, note that here, more definite than "assume"

Reply:
Yes, will clarify